src/tasks/DockerPseudoterminal.ts

```diff
@@ -56,33 +62,52 @@ export class DockerPseudoterminal implements Pseudoterminal {
     public async executeCommandInTerminal(
         command: CommandLineBuilder,
         folder: WorkspaceFolder,
-        rejectOnStderr?: boolean,
-        stdoutBuffer?: Buffer,
-        stderrBuffer?: Buffer,
         token?: CancellationToken): Promise<void> {
         const commandLine = resolveVariables(command.build(), folder);
 
         // Output what we're doing, same style as VSCode does for ShellExecution/ProcessExecution
         this.write(`> ${commandLine} <\r\n\r\n`, DEFAULTBOLD);
 
-        // TODO: Maybe support remote Docker hosts and do addDockerSettingsToEnvironment?
-        await spawnAsync(
-            commandLine,
-            { cwd: folder.uri.fsPath },
-            (stdout: string) => {
-                this.writeOutput(stdout);
-            },
-            stdoutBuffer,
-            (stderr: string) => {
-                this.writeError(stderr);
-
-                if (rejectOnStderr) {
-                    throw new Error(stderr);
+        const pty = getCoreNodeModule<nodepty>('node-pty');
+
+        this.activePty = pty.spawn(isWindows() ? 'powershell.exe' : 'sh', ['-c', commandLine], {

```

Comment:
Should we be using the shell (default or otherwise) set by the user?  (I prefer and setup `CMD` on Windows, for example, and `zsh` on Mac.). Is PowerShell now a core component of Windows or still an optional feature?

Reply:
Probably should use `cmd` and `sh`, rather than `powershell` and `sh`. We *could* try to adopt the user's shell choice but that becomes *really* complicated *really* quickly. In defense of just using `cmd`/`sh`, that's what Node does if you do `child_process.spawn`, and since `spawn` is what we were doing previously--that's what we've always had.

Reply:
I'm fine with `cmd` and `sh`; I think that should work reasonably well in the vast majority of cases.

Reply:
Ugh. Obnoxiously, `node-pty` does not handle arguments correctly with `cmd`. Something with quoting is not working.

Reply:
So it's just not plausible to use `node-pty`. There is so much complexity in handling the terminal control characters that it's not worth the effort. Instead I'll just make stderr text be default color. Thanks for nothing, BuildKit!